src/python/pants/backend/python/tasks2/setup_py.py

```diff
@@ -607,10 +611,10 @@ def create(target):
           self.context.log.info('Writing {}'.format(sdist_path))
           shutil.move(setup_runner.sdist(), sdist_path)
           safe_rmtree(setup_dir)
-          executed[target] = sdist_path
+          python_dists.add(target, os.path.dirname(sdist_path)).append(os.path.basename(sdist_path))
         else:
           self.context.log.info('Running {} against {}'.format(self._run, setup_dir))
           setup_runner = SetupPyRunner(setup_dir, self._run, interpreter=interpreter)
           setup_runner.run()
-          executed[target] = setup_dir
-    return executed
+          python_dists.add(target, os.path.dirname(setup_dir)).append(os.path.basename(setup_dir))
+    return python_dists  # returned for tests
```

Comment:
Not your fault, but this was always yucky. Can you add a TODO to modify the tests to get this data from the products, and remove this return?